crates/noirc_evaluator/src/ssa/integer.rs

```diff
@@ -257,158 +237,108 @@ pub fn block_overflow(
             ins.operator,
-            node::Operation::Nop
-                | node::Operation::Call(_)
-                | node::Operation::Res
-                | node::Operation::Ret
+            Operation::Nop | Operation::Call(..) | Operation::Results { .. } | Operation::Return(_)
         ) {
             //For now we skip completely functions from overflow; that means arguments are NOT truncated.
             //The reasoning is that this is handled by doing the overflow strategy after the function has been inlined
             continue;
         }
-        let mut ins_args = Vec::new();
-        let mut i_lhs = ins.lhs;
-        let mut i_rhs = ins.rhs;
+
         //we propagate optimised loads - todo check if it is needed because there is cse at the end
-        if node::is_binary(ins.operator) {
-            //binary operation:
-            i_lhs = optim::propagate(ctx, ins.lhs);
-            i_rhs = optim::propagate(ctx, ins.rhs);
-        }
         //We retrieve get_current_value() in case a previous truncate has updated the value map
-        let r_id = get_value_from_map(i_rhs, &value_map);
-        let mut update_instruction = false;
-        if r_id != ins.rhs {
-            ins.rhs = r_id;
-            update_instruction = true;
-        }
-        let l_id = get_value_from_map(i_lhs, &value_map);
-        if l_id != ins.lhs {
-            ins.lhs = l_id;
-            update_instruction = true;
-        }
-        let r_obj = ctx.try_get_node(r_id);
-        let l_obj = ctx.try_get_node(l_id);
-        let r_max = get_obj_max_value(ctx, r_obj, r_id, max_map, &value_map);
-        let l_max = get_obj_max_value(ctx, l_obj, l_id, max_map, &value_map);
-        //insert required truncates, except for field type or dummy node
-        let to_truncate = ins.truncate_required(get_size_in_bits(l_obj), get_size_in_bits(r_obj));
-        if to_truncate.0 && l_obj.is_some() && get_type(l_obj) != node::ObjectType::NativeField {
-            //adds a new truncate(lhs) instruction
-            add_to_truncate(ctx, l_id, get_size_in_bits(l_obj), &mut truncate_map, max_map);
-        }
-        if to_truncate.1 && r_obj.is_some() && get_type(r_obj) != node::ObjectType::NativeField {
-            //adds a new truncate(rhs) instruction
-            add_to_truncate(ctx, r_id, get_size_in_bits(r_obj), &mut truncate_map, max_map);
-        }
+        let should_truncate = ins.truncate_required();
+        let ins_max_bits = get_instruction_max(ctx, &ins, max_map, &value_map).bits();
+        let res_type = ins.res_type;
+
+        let too_many_bits = ins_max_bits >= FieldElement::max_num_bits() as u64
+            && res_type != ObjectType::NativeField;
+
+        ins.operator.map_id_mut(|id| {
+            let id = optim::propagate(ctx, id);
+            let id = get_value_from_map(id, &value_map);
+
+            get_obj_max_value(ctx, id, max_map, &value_map);
+            let obj = ctx.try_get_node(id);
+            let should_truncate_obj =
```

Comment:
It is needed, the if condition is too long otherwise so this improves readability by separating the clauses.

Reply:
I disagree, it is less readable like this, although it is a matter of style and preference.
Here you have to substitute the variable in your head instead of having everything explicit, the long clause can then be separated with a new line.
I think that `too_many_bits `and `should_truncate` are helpful and meaningful, but `should_truncate_obj` is not clear and I prefer not to have it.
If I did not convince you or if you do not want to do a change just for this, fell free to simply resolve this comment, I won't re-open it!

Reply:
I agree that `should_truncate_obj` is not the best name, I've tried to change around the names a bit to improve them but honestly did not improve much.